packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -574,6 +574,12 @@ class _AndroidMotionEventConverter {
       return null;
     }
 
+    // These arrays must have same length, otherwise below 'AndroidMotionEvent' constructor
+    // will throw an exception.
+    if (pointers.length != pointerProperties.length || pointers.length != pointerPositions.length) {
```

Comment:
If these arrays must have the same length, we should assert that here, rather than ignoring them when they don't, and then we should go into the relevant engine code (flutter/engine repo) and make sure we never generate lists with inconsistent lengths.